src/packages/core/src/servers/ws-server.ts

```diff
@@ -19,6 +20,48 @@ type WebSocketCapableFlavorMap = {
     : never;
 };
 
+export function sendFragmented(
+  ws: WebSocket,
+  data: Generator<Buffer, void, void>,
+  useBinary: boolean,
+  chunkSize: number
+) {
+  ws.cork(() => {
+    // fragment send: https://github.com/uNetworking/uWebSockets.js/issues/635
+    const shouldCompress = false;
+
+    const fragments = getFragmentGenerator(data, chunkSize);
+    // get our first fragment
+    const { value: firstFragment } = fragments.next();
+    // check if there is any more fragments after this one
+    let { value: maybeLastFragment, done } = fragments.next();
+    // if there are no more fragments send the "firstFragment" via `send`, as
+    // we don't need to chunk it.
+    if (done) {
+      ws.send(firstFragment as RecognizedString, useBinary);
+    } else {
+      // since we have at least two fragments send the first one now that it
+      // is "full"
+      ws.sendFirstFragment(firstFragment as RecognizedString, useBinary);
+      // at this point `maybeLastFragment` is the next fragment that should be
```

Comment:
I could try to suggest corrections on this comment block but I think it would be easier if you went over it :laughing: I get the gist of what you're saying, but there definitely are some typos

Reply:
Huh, I don't recall having a stroke whilst writing that comment. ¯\\_(ツ)_/¯

I've degibberished the comment so hopefully it makes sense now!